decorators_library/decorators.py

```diff
@@ -1 +1,89 @@
-# implement your decorators here.
+import signal
+from .exceptions import TimeoutError
+import logging
+import time
+
+def timeout_occured(signum, frame):
+    if timeout.running:
+        raise TimeoutError('Function call timed out')
+    
+def timeout(time):
+    def timeout_decorator(func):
+        def run_timeout(*args, **kwargs):
+            signal.signal(signal.SIGALRM, timeout_occured)
+            signal.alarm(time)
+            timeout.running = True
+            result = func(*args, **kwargs)
+            timeout.running = False
+            return result
+        return run_timeout
+    return timeout_decorator
+            
+def debug(logger=None):
+    def logging_decorator(func):
+        def log_function(*args, **kwargs):
+            if not logger:
+                debug_logger = logging.getLogger(func.__module__)
+            else:
+                debug_logger = logger
+            debug_logger.debug('Executing "{}" with params: {}, {}'.format(func.__name__, args, kwargs))
+            result = func(*args, **kwargs)
+            debug_logger.debug('Finished "{}" execution with result: {}'.format(func.__name__, result))
+            return result
+        return log_function
+    return logging_decorator
+
+class count_calls():
+    counters_dict = {}
+    
+    def __init__(self, func):
+        self.func = func
+        self.counters_dict[func.__name__] = 0
+        
+    def __call__(self, *args, **kwargs):
```

Comment:
we also used `**kwargs` iirc are still not sure whether it is necessary. wonder whether it's necessary, hm hm